pkgs/top-level/packages-config.nix

```diff
@@ -6,38 +6,33 @@
   # Enable recursion into attribute sets that nix-env normally doesn't look into
   # so that we can get a more complete picture of the available packages for the
   # purposes of the index.
-  packageOverrides = super:
-  let
-    recurseIntoAttrs = sets:
-      super.lib.genAttrs
-        (builtins.filter (set: builtins.hasAttr set super) sets)
-        (set: super.recurseIntoAttrs (builtins.getAttr set super));
-  in recurseIntoAttrs [
-    "roundcubePlugins"
-    "emscriptenfastcompPackages"
-    "fdbPackages"
-    "nodePackages_latest"
-    "nodePackages"
-    "platformioPackages"
-    "haskellPackages"
-    "idrisPackages"
-    "sconsPackages"
-    "gns3Packages"
-    "quicklispPackagesClisp"
-    "quicklispPackagesSBCL"
-    "rPackages"
-    "apacheHttpdPackages_2_4"
-    "zabbix50"
-    "zabbix40"
-    "zabbix30"
-    "fusePackages"
-    "nvidiaPackages"
-    "sourceHanPackages"
-    "atomPackages"
-    "emacs26.pkgs"
-    "emacs27.pkgs"
-    "steamPackages"
-    "ut2004Packages"
-    "zeroadPackages"
-  ];
+  packageOverrides = super: with super; lib.mapAttrs (n: v: recurseIntoAttrs v) {
+    inherit
+      roundcubePlugins
+      fdbPackages
+      nodePackages_latest
+      nodePackages
+      platformioPackages
+      haskellPackages
+      idrisPackages
+      sconsPackages
+      gns3Packages
+      quicklispPackagesClisp
+      quicklispPackagesSBCL
+      rPackages
+      apacheHttpdPackages_2_4
+      zabbix50
+      zabbix40
+      zabbix30
+      fusePackages
+      sourceHanPackages
+      atomPackages
+      steamPackages
+      ut2004Packages
+      zeroadPackages
+    ;
+      # inherit (emacs) pkgs;
```

Comment:
Found the error, this results in `pkgs = emacs.pkgs`, so I'm actually setting overriding `pkgs` with `emacs.pkgs`

Whoops.

This will require some more thought but feel free to steal the idea.

Reply:
Instead of using `inherit` we could simply say:
`emacs26.pkgs = emacs26.pkgs`

It looks a bit silly but it works.

Reply:
Yeah, that's what I've been thinking too but I'm still looking for an even cleaner solution.  
Handling nested subsets separately like this would be good enough though.

I think we'd also need to set recurseIntoAttrs on emacs though.

Reply:
As `emacs.pkgs = {}` expands to `{ emacs = { pkgs = {}; }; }` the outer attrset is already caught by the `mapAttrs` call.